acapy_agent/anoncreds/default/legacy_indy/registry.py

```diff
@@ -1229,3 +1231,18 @@ async def txn_submit(
                 )
         except LedgerError as err:
             raise AnonCredsRegistrationError(err.roll_up) from err
+
+    async def get_schema_info_by_id(self, schema_id: str) -> AnoncredsSchemaInfo:
```

Comment:
@jamshale I think we should use the existing methods

- get_schema
- get_credential_definition

instead of these new methods as it's not guaranteed that the name, version or other attributes required are within the identifiers. For cheqd we have to make a ledger query inorder to get them. The above functions already handle this. I created a reference PR [here](https://github.com/jamshale/acapy/pull/178)

Reply:
That had been my original thought, but this is used only by the `holder`. The holder shouldn't necessarily need to go to the ledger or wallet to get the issuer id and schema version, etc., if the information is available in the schema id and cred def id itself.

For `cheqd` this `get_schema_info_by_id` method could call the the get_schema method and extract the correct info for the holder if it's required that the holder needs to contact the ledger to get the required info.

I'm not sure what did methods will be able to put the required info into the schema and cred_def id's themselves.

It'd be nice to get some feedback from others on this...

Reply:
I could change the handler method implementations like you did as these are done by the `verifier` and they should be able to contact the ledger. My comment more applies to the usage in https://github.com/openwallet-foundation/acapy/blob/main/acapy_agent/anoncreds/holder.py#L224 when the holder save the credential.

I could remove the `get_cred_def_info_by_id` method by changing the handler like you did.

Reply:
I updated this to only use the `get_schema_info_by_id` method with the holder when saving the credential.

Reply:
In that case the profile should be one of the parameters within get_schema_info_by_id.

Reply:
I thought it would be available through the Registry class, but looks like all the other functions pass it as a parameter so i'll follow the pattern.